pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -745,13 +781,71 @@ func (p *PriorityQueue) AssignedPodUpdated(pod *v1.Pod) {
 // if Pop() is waiting for an item, it receives the signal after all the pods are in the
 // queue and the head is the highest priority pod.
 func (p *PriorityQueue) moveAllToActiveOrBackoffQueue(event framework.ClusterEvent, preCheck PreEnqueueCheck) {
-	unschedulablePods := make([]*framework.QueuedPodInfo, 0, len(p.unschedulablePods.podInfoMap))
+	activated := false
+
 	for _, pInfo := range p.unschedulablePods.podInfoMap {
-		if preCheck == nil || preCheck(pInfo.Pod) {
-			unschedulablePods = append(unschedulablePods, pInfo)
+		isSchedulable := framework.PodMaybeSchedulable
+		if preCheck != nil {
+			isSchedulable = preCheck(pInfo.Pod)
+		}
+
+		if isSchedulable == framework.PodNotAffected {
+			continue
+		}
+
+		// If the event doesn't help making the Pod schedulable, then keep it in its current state.
+		// Note: we don't run the check if pInfo.UnschedulablePlugins is nil, which denotes
+		// either there is some abnormal error, or scheduling the pod failed by plugins other than PreFilter, Filter and Permit.
+		// In that case, it's desired to move it anyways.
+		if len(pInfo.UnschedulablePlugins) != 0 && !p.podMatchesEvent(pInfo, event) {
+			continue
+		}
+
+		pod := pInfo.Pod
+		if isSchedulable != framework.PodImmediatelySchedulable && p.isPodBackingoff(pInfo) {
+			if err := p.podBackoffQ.Add(pInfo); err != nil {
+				klog.ErrorS(err, "Error adding pod to the backoff queue", "pod", klog.KObj(pod))
+			} else {
+				klog.V(5).InfoS("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", event, "queue", backoffQName)
+				metrics.SchedulerQueueIncomingPods.WithLabelValues("backoff", event.Label).Inc()
+				p.unschedulablePods.delete(pod, pInfo.Gated)
+			}
+		} else {
+			gated := pInfo.Gated
```

Comment:
Is it OK to ignore backoff this case? 
I think the scheduling queue wants to honor backoff time in any cases. 

What if plugin returns `PodImmediatelySchedulable` to many Pods and many Pods goes to activeQ without backoff? Doesn't that affect badly to other Pod's scheduling?